src/main/java/com/nike/cerberus/AssetResourceFileFactory.java

```diff
@@ -0,0 +1,80 @@
+package com.nike.cerberus;
+
+import com.google.common.collect.ImmutableMap;
+import com.nike.cerberus.domain.AssetResourceFile;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+/**
+ * Factory for creating {@link AssetResourceFile} objects
+ */
+public class AssetResourceFileFactory {
+    private static final Logger logger = LoggerFactory.getLogger(AssetResourceFileFactory.class);
+
+    private static final String DEFAULT_MIME_TYPE = "text/plain";
+
+    private static final ImmutableMap<String, String> FILE_EXT_TO_MIME_TYPE_MAP = ImmutableMap.<String, String>builder()
+            .put("svg", "image/svg+xml")
+            .put("js", "text/javascript")
+            .put("ico", "image/x-icon")
+            .put("css", "text/css")
+            .put("html", "text/html")
+            .put("json", "application/json")
+            .put("jpeg", "image/jpeg")
+            .put("jpg", "image/jpeg")
+            .put("png", "image/png")
+            .put("gif", "image/gif")
+            .put("pdf", "application/pdf")
+            .build();
+
+    public static AssetResourceFile create(String filename, String filePath, String rootFolderPath) {
+        return new AssetResourceFile(
+                filename,
+                getRelativePath(filePath, rootFolderPath),
+                getMimeTypeForFileFromName(filename),
+                getFileContents(filePath));
+    }
+
+    /**
+     * @return The contents of the given file in bytes
+     */
+    private static byte[] getFileContents(String filePath) {
+        try {
+            ClassLoader loader = AssetResourceFileFactory.class.getClassLoader();
+            return IOUtils.toByteArray(loader.getResourceAsStream(filePath));
+        } catch (NullPointerException | IOException ioe) {
+            logger.info("Could not read contents of file: " + filePath, ioe);
```

Comment:
Is logger.error better here?  And are you sure we want to swallow exceptions here?

Reply:
@tlisonbee No, that was a mistake. Thanks